x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/plugin.ts

```diff
@@ -0,0 +1,146 @@
+import { setSavedObjectsClient } from './application/components/builtin_alert_types/threshold/lib/api';
+import { hasShowActionsCapability, hasShowAlertsCapability } from './application/lib/capabilities';
+
+export type Setup = void;
+export type Start = void;
+
+const REACT_ROOT_ID = 'triggersActionsRoot';
+
+export class Plugin implements CorePlugin<Setup, Start> {
+  private actionTypeRegistry: ActionTypeRegistry;
+  private alertTypeRegistry: AlertTypeRegistry;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    const actionTypeRegistry = new ActionTypeRegistry();
+    this.actionTypeRegistry = actionTypeRegistry;
+
+    const alertTypeRegistry = new AlertTypeRegistry();
+    this.alertTypeRegistry = alertTypeRegistry;
+  }
+
+  public setup(core: CoreSetup, plugins: any): Setup {
+    /*
+      The code below would be replacing for current:
+      uiExports: {
+        managementSections: ['myplugin/management'],
+      }
+    */
+    const {
+      capabilities,
+      management: { getSection },
+    } = plugins;
+
+    const canShowActions = hasShowActionsCapability(capabilities.get());
+    const canShowAlerts = hasShowAlertsCapability(capabilities.get());
+    if (canShowActions || canShowAlerts) {
+      registerBuiltInActionTypes({
+        actionTypeRegistry: this.actionTypeRegistry,
+      });
+
+      registerBuiltInAlertTypes({
+        alertTypeRegistry: this.alertTypeRegistry,
+      });
+
+      const kbnSection = getSection('kibana');
+      kbnSection.register('triggersActions', {
+        display: i18n.translate('xpack.triggersActionsUI.managementSection.displayName', {
+          defaultMessage: 'Alerts and actions',
+        }),
+        order: 7,
+        url: `#${BASE_PATH}`,
+      });
+    }
+  }
+
+  public start(core: CoreStart, plugins: any) {
+    const { capabilities } = plugins;
+    const canShowActions = hasShowActionsCapability(capabilities.get());
+    const canShowAlerts = hasShowAlertsCapability(capabilities.get());
+    // AppCore/AppPlugins to be passed on as React context
+    const AppDependencies = {
+      core,
+      plugins,
+      actionTypeRegistry: this.actionTypeRegistry,
+      alertTypeRegistry: this.alertTypeRegistry,
+    };
+
+    // Don't register routes when user doesn't have access to the application
+    if (!canShowActions && !canShowAlerts) {
+      return;
+    }
+
+    docTitleService.init(plugins.docTitle.change);
+    breadcrumbService.init(core.chrome, plugins.management.breadcrumb);
+
+    const unmountReactApp = (): void => {
+      const elem = document.getElementById(REACT_ROOT_ID);
+      if (elem) {
+        unmountComponentAtNode(elem);
+      }
+    };
+
+    routes.when(`${BASE_PATH}/:section?/:subsection?/:view?/:id?`, {
+      template,
+      controller: (() => {
+        return ($route: any, $scope: any, Private: any) => {
+          const appRoute = $route.current;
+          setSavedObjectsClient(Private(SavedObjectsClientProvider));
```

Comment:
If this `api.ts` module requires an instance of the SavedObjectsClient, it would probably be better to change it to return a class or factory function that takes the SavedObjectsClient as a param. Then you could create an instance of that object and add it to your React application context. Another option would be to remove the state from the module entirely and just require the SavedObjectsClient to be passed in when calling your APIs functions, but that's probably more cumbersome if it is used in a lot of places.